### DISCUSSION TITLE
RAR-Cask: where to place registration-data (rarreg.key)? Only working in /etc/?

### DISCUSSION THREAD

**Discussion by borque** [Homebrew/discussions#2880] [Casks]
Hello!
I recently installed version 6.1 of the rar-cask. Coming from windows I have valid registration data in form of a file called rarreg.key. In "normal" installations of RAR in Linux- or Mac- environments this file needs to be placed into /etc/.

As the other rar file that usually needs to go to /etc/, rarfiles.lst, is installed to /opt/homebrew/etc by the rar-cask, I tried to place my rarreg.key there, too. But it is not recognized there or in the directory next to the binaries. Only when placed in /etc/ (i.e. the systems /etc directory) rar says it is registered.

This brings up two questions:
1) Is there anything I am making wrong? Where might be the right place for rarreg.key within /opt/homebrew to be recognized by rar?
2) Are the two other files, rarfiles.lst, which resides in /opt/homebrew/etc, and default.sfx which resides in /opt/homebrew/lib, in the right place to be used by homebrews rar-installation or do they also need to be placed into the corresponding system directories?

Thanks for help!

**Comment by SMillerDev** [comment]
Casks are just running the upstream installer do they don't respect homebrew prefixes. Whatever upstream documents is the right location

**Reply by carlocab** [reply]
In this case there is no upstream installer though; just a tarball. So this cask decides where to put things.

**Comment by neutric** [comment]
Wouldn't the path where the binary looks for stuff like this be determined by upstream as they are the ones compiling it? 

Quoting `rar.txt` from within the tar.gz:

> RAR and UnRAR for Unix read configuration information from .rarrc file in a user's home directory (stored in HOME environment variable) or in /etc directory.

`/etc` unless you tell it otherwise through `.rarrc` seems to match what you described.

Regarding your second question, specifically `rarfiles.lst`, again quoting `rar.txt`:

> In Windows this file should be placed in the same directory as RAR or in %APPDATA%\WinRAR directory, in Unix - to the user's home directory or in /etc.

No clue if and how `#{HOMEBREW_PREFIX}/etc` is interwoven with the system defaults, but given your experience with the license file I'd guess `rar` won't pick it up, either. 

You could try it out by modifying `rarfiles.lst` and comparing the resulting rar-archives. In `rarfiles.lst` it says 

> Here you may modify the file list which tells RAR the order in which to add files to a solid archive.

and

> Tips to provide improved compression and speed of operation:
>  - files containing similar information should be grouped together in the archive if possible

which should result in different archive sizes depending on the file order you specify.

**Comment by borque** [comment]
I could not find out about rarfiles.lst, but default.sfx isn‘t found by rar in the directory where it is installed by this cask. The same applies to rarreg.key as mentioned above. So I strongly suspect the same applies to rarfiles.lst.
As a solution I created softlinks (ln -s) in /etc and /usr/local/lib which point to the files installed to /opt/… .
This is working for now and whenever rar (and the cask) get updated by brew the links will simply point to the new files.
